src/explain.rs

```diff
@@ -1170,13 +1043,103 @@ impl<L: Language> Explain<L> {
         equalities
     }
 
-    pub(crate) fn populate_enodes<N: Analysis<L>>(&self, mut egraph: EGraph<L, N>) -> EGraph<L, N> {
-        for i in 0..self.explainfind.len() {
-            let node = &self.explainfind[i];
-            egraph.add(node.node.clone());
+    pub(crate) fn with_nodes<'a>(&'a mut self, nodes: &'a [L]) -> ExplainNodes<'a, L> {
```

Comment:
`with_nodes` is a little awkward.
Another way to implement would be to keep the nodes in the `Explain` struct from the start. I think that might be cleaner?

Reply:
The nodes need to be available when explanations are disabled so I couldn't only have them in the `Explain` struct like they were before, I didn't want to duplicate all of the nodes by having them in both places since that would require extra unnecessarily memory, and since rust doesn't support self-referential structs, I thought this was the easiest way to not change the explain implementation to much.

Reply:
Hmm, perhaps the Explain struct should be around whether explanations are disabled or enabled? Or I supposed the other fields of Explain could be wrapped in an `Option<OtherExplanationFields>`

Reply:
Do you mean have
```rust
struct Explain {
  nodes: Vec<L>,
  rest: Option<OtherExplanationFields>,
}
```
I don't think this would simplify things very much, and the nodes are use outside of explanations as well so I don't see why they should be part of the explain struct.
Some of the explain methods also take in the unionfind and classes, so I could also thread the nodes as an extra parameter if you prefer.

Reply:
Yeah, I see what you mean.
I don't have strong feelings, the solution you have is not too bad

Reply:
If you're fine with it, I think I'll just leave it as it is. Is there anything you would like me to change before I merge this?

Reply:
Nope, everything looks good to me

Reply:
Great, Thanks